src/main/java/bio/terra/workspace/service/datarepo/DataRepoService.java

```diff
@@ -31,12 +31,12 @@ private RepositoryApi repositoryApi(String instance, AuthenticatedUserRequest us
   }
 
   public void validateInstance(String instance) {
-    if (!dataRepoConfig.getInstances().contains(instance)) {
+    if (!dataRepoConfig.getInstances().containsValue(instance.toLowerCase().trim())) {
```

Comment:
this should be `containsKey` - the data references should be storing the string `terra-dev` or `terra-prod`